Reset/test

hyde777 opened the pull request:
I feel like i lack stuff on my test.
What are your thought?

Commit:
Started some test

We should talk about which exception we would use.

Commit:
Added Bones of Resetor

Commit:
Rename Forgetor to Resetor.

# Conflicts:
#	src/main/java/fr/esgi/simple_auth_java/reset/ResetException.java
#	src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMail.java

Commit:
Exception -> RuntimeException

Added some method

Commit:
update Gitignore

alex-lairan committed:
Create Auth method for user

alex-lairan committed:
Implement SMS authentication

alex-lairan committed:
Update tests

alex-lairan committed:
Connect user when okay

alex-lairan committed:
Move connection into Manager class

alex-lairan committed:
Update tests manager to pass

Commit:
Modified Tests

TristanEduProjet commented on src/main/java/fr/esgi/simple_auth_java/reset/BadPasswordMatchException.java:
```diff
@@ -0,0 +1,7 @@
+/**
+ * Created by Yohan FAIRFORT on 10/06/2017.
+ */
+public class BadPasswordMatchException extends RuntimeException {
```
c'est pour quoi cette classe ?

TristanEduProjet replied:
regarde `IllegalResetException` ou `ResetException`

hyde777 replied:
Le truc c'est que pour moi IllegalResetException et ResetException ne sont pas assez explicite

Je ne sais même pas dans quel cas elles se trigger

TristanEduProjet replied:
la javadoc n'est peut-être pas assez explicite
`IllegalResetException` survient si l'utilisateur à posé un problème lors du reset, et `ResetException` est pour englober toutes les autres exceptions qui auraient pu se produire (`IOException`, `SQLException`, `IndexOutOfBoundException`, etc)

hyde777 replied:
Bah c'est bizarre alors: on a aucun moyen de savoir si il y a eu des problème dans le reset autre que les problème générique?

Genre si le nouveau password est trop faible, comment est-ce qu'on le remonte? si on envoie un mail et qu'il ne répond jamais comment est-ce qu'on le sait? C'est ce genre d'exception qu'il nous manque.

TristanEduProjet replied:
le mot de passe trop faible est aussi tester qu'en test unitaire pour l'instant (on a pas eu le temps de faire toute la partie des contraintes)

TristanEduProjet replied:
pour le mail, c'est obligatoire dans la procédure actuel du resetor, car il attend le retour de l'utilisateur, donc obliger qu'il réponde, sinon attente infinie
donc pas d'exception pour ca pour l'instant

hyde777 replied:
ha! Dans ce cas c simple: si le mot de passe est trop simple tu envoie une exception custom ( en RuntimeException pas besoin de try catch)

hyde777 replied:
Pas besoin de contrainte y'a juste besoin du cas pour

TristanEduProjet replied:
non ne t'occupe pas de la force du mot de passe, oublie ça
c'est pas au resetor de s'en occuper, c'est au user ; on modifiera ce qu'il faut quand ce sera mis en place ;)

hyde777 replied:
Tu veux dire que si le nouveau mot de passe est trop faible c'est le User qui doit dire qu'il est trop faible? sa me semble pas logique: c'est le resetor qui fait l'action de reset le mdp, le mdp ne devrait même pas atteindre le User en cas de problème dans le Resetor.

TristanEduProjet replied:
bah si c'est logique
mais bon on est à court de temps, alors t'en occupe pas, il n'y a pas cette vérification dans le projet pour l'instant

TristanEduProjet commented on src/main/java/fr/esgi/simple_auth_java/reset/WeakPasswordException.java:
```diff
@@ -0,0 +1,7 @@
+/**
+ * Created by Yohan FAIRFORT on 10/06/2017.
+ */
+public class WeakPasswordException extends RuntimeException {
```
idem que l'autre, pourquoi as-tu cette classe ?

TristanEduProjet commented on src/main/java/fr/esgi/simple_auth_java/reset/WrongMailException.java:
```diff
@@ -0,0 +1,7 @@
+/**
+ * Created by Yohan FAIRFORT on 10/06/2017.
+ */
+public class WrongMailException extends RuntimeException {
```
tu fais une vérification ?
c'est au User de s'en occuper

hyde777 replied:
mhhh plutot au mailer non?

TristanEduProjet replied:
non, au user, car c'est une contrainte propre à lui
d'ailleurs c'est un point qui sera ajouter plus tard (elle est partiellement mise en place)

hyde777 replied:
Heu non non, je parle pas de si le mail est correcte, mais si l'envoie d'un message via le mail marche vraiment.(un ping?)

TristanEduProjet replied:
j'ai utilisé une fonctionnalité spéciale de la lib pour rediriger en dev vers les log qui pointe sur la console
je t'ai laissé exprès une fonction pour modifier le mailer du resetor pour que tu puisse le remplacer par un mock et ainsi capturé le token ou mdp temporaire dans tes tests

hyde777 replied:
C'est quoi la fonction? (faudrais l'ajouter dans l'interface alors non?)

TristanEduProjet replied:
non, car le mail est spécifique au resetor qu'il l'utilise, et bonne chance pour faire une interface générique pour (sans être exhaustif) Console + Mail + SMS + HTTP ...

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetTest.java:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Created by Yohan FAIRFORT on 06/06/2017.
+ */
+public class ResetTest {
```
le nom de la classe est incorrect
le nom doit être nomDeLaClasseTestée + Test

hyde777 replied:
Je vais delete cette classe, oublié de la faire

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetTest.java:
```diff
@@ -0,0 +1,27 @@
+public class ResetTest {
+
+    @Mock User testUser;
+    Manager testManager = new Manager();
```
pourquoi le manager ?
tu test le Resetor qui n'a pas besoin du manager (interface externe)

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetTest.java:
```diff
@@ -0,0 +1,27 @@
+    @Mock User testUser;
+    Manager testManager = new Manager();
+
+    Resetor resetPasswordWithTempPwdByMail = mock(ResetorPasswordWithTempPwdByMail.class);
```
t'as fait l'inverse, c'est `ResetorPasswordWithTempPwdByMail` que tu dois tester, donc pas un mock

hyde777 replied:
Ok

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetTest.java:
```diff
@@ -0,0 +1,27 @@
+    @Test
+    public void should_success()
+    {
+        testManager.reset(testUser, resetPasswordWithTempPwdByMail);
```
tu exécute le reset, mais où est-ce que tu teste que le reste s'est bien déroulé ?

hyde777 replied:
C sa ce que je comprenais pas, comment je sais que le reset s'est bien passer?

TristanEduProjet replied:
vu que le user est un mock, tu peux savoir s'il à était modifié par le resetor
tu test en plus que le nouveau mot de passe est bien celui que tu as voulu
(avec l'encryptor-disable)

hyde777 replied:
heu on a pas juste besoin de tester si le nouveau mail est différent de l'ancien?

TristanEduProjet replied:
bah non, vu que c'est le mdp qui est changé

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMailTest.java:
```diff
@@ -0,0 +1,50 @@
+        String oldPass = testUser.getPassword();
+        testManager.reset(testUser, resetPasswordByMail);
+        try {
+            TimeUnit.MINUTES.sleep(30);
```
il n'y a pas de limite de temps dans les resetor

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMailTest.java:
```diff
@@ -0,0 +1,50 @@
+    User testUser;
+    Manager testManager = new Manager();
+
+    Resetor resetPasswordByMail = mock(ResetorPasswordByMail.class);
```
idem

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMailTest.java:
```diff
@@ -0,0 +1,50 @@
+    public void should_fail_because_wrong_mail()
+    {
+        testUser.setEmail(null);
+        testManager.reset(testUser, resetPasswordByMail);
```
idem

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithTokenTest.java:
```diff
@@ -0,0 +1,43 @@
+    User testUser;
+    Manager testManager = new Manager();
+
+    Resetor resetPasswordWithToken = mock(ResetorPasswordWithOldPwd.class);
```
idem

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithTokenTest.java:
```diff
@@ -0,0 +1,43 @@
+    @Test(expected = ResetException.class)
+    public void should_fail_because_wrong_token()
+    {
+        testManager.reset(testUser, resetPasswordWithToken);
```
n'échouera pas, et ou sont les input de test ?
parce que là sinon ton test va bloquer (le resetor attend l'input), et ça n’avancera plus

hyde777 replied:
Ha! il attend quel type d'input?

TristanEduProjet replied:
`System.in`
regarde sur la branche de @Maill il à fait des choses sur ca

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithTokenTest.java:
```diff
@@ -0,0 +1,43 @@
+public class ResetorPasswordWithTokenTest {
+    @Mock
+    User testUser;
+    Manager testManager = new Manager();
```
idem

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwdTest.java:
```diff
@@ -0,0 +1,50 @@
+    public void should_fail_because_weak_new_password()
+    {
+        testUser.setPassword("weak");
+        testManager.reset(testUser, resetPasswordByMail);
```
et les inputs ?

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwdTest.java:
```diff
@@ -0,0 +1,50 @@
+    @Test(expected= BadPasswordMatchException.class)
+    public void should_fail_because_bad_password()
+    {
+        testManager.reset(testUser, resetPasswordByMail);
```
idem

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwdTest.java:
```diff
@@ -0,0 +1,50 @@
+public class ResetorPasswordWithOldPwdTest {
+    @Mock
+    User testUser;
+    Manager testManager = new Manager();
```
pas nécessaire

TristanEduProjet commented on src/test/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwdTest.java:
```diff
@@ -0,0 +1,50 @@
+    User testUser;
+    Manager testManager = new Manager();
+
+    Resetor resetPasswordByMail = mock(ResetorPasswordByMail.class);
```
idem

TristanEduProjet reviewed (changes requested):
beaucoup de changements

Commit:
Test in progress

TristanEduProjet commented on src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwd.java:
```diff
@@ -5,14 +5,14 @@
  * Created by Blixel on 04/06/2017.
  */
-public class ForgetorPasswordByMail implements Forgetor {
+public class ResetorPasswordWithOldPwd implements Resetor {
```
mauvais merge ...
`ForgetorPasswordWithTempPwdByMail` --> `ResetorPasswordByMail` --> `ResetorPasswordWithOldPwd` --> `ResetorPasswordWithTempPwdByMail` ??

TristanEduProjet commented on src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMail.java:
```diff
@@ -1,18 +1,18 @@
  * Created by Blixel on 04/06/2017.
  */
-public class ForgetorPasswordWithTempPwdByMail implements Forgetor {
+public class ResetorPasswordByMail implements Resetor {
```
mauvais merge ...
`ForgetorPasswordWithTempPwdByMail` --> `ResetorPasswordByMail` --> `ResetorPasswordWithOldPwd` --> `ResetorPasswordWithTempPwdByMail` ??

TristanEduProjet commented on src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithTempPwdByMail.java:
```diff
@@ -1,18 +1,19 @@
  * Created by Blixel on 04/06/2017.
  */
-public class ForgetorPasswordWithOldPwd implements Forgetor {
+public class ResetorPasswordWithTempPwdByMail implements Resetor {
```
mauvais merge ...
`ForgetorPasswordWithTempPwdByMail` --> `ResetorPasswordByMail` --> `ResetorPasswordWithOldPwd` --> `ResetorPasswordWithTempPwdByMail` ??

hyde777 replied:
Dude c'est le pb que je te parlais hier, Git est juste complètement perdu xD.

TristanEduProjet replied:
bon je vais te résoudre ça (entorse à la règle)

TristanEduProjet reviewed (changes requested):
problèmes de ton merge

TristanEduProjet commented:
pour les conflits suivants :
- src/main/java/fr/esgi/simple_auth_java/Manager.java
- src/main/java/fr/esgi/simple_auth_java/reset/ResetException.java
- src/main/java/fr/esgi/simple_auth_java/reset/Resetor.java
- src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMail.java
- src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwd.java

tu choisi "utiliser le leurs", la version de la common, ça évitera d'écraser la bonne version

TristanEduProjet committed:
update from develop

# Conflicts:
#	src/main/java/fr/esgi/simple_auth_java/Manager.java
#	src/main/java/fr/esgi/simple_auth_java/reset/IllegalResetException.java
#	src/main/java/fr/esgi/simple_auth_java/reset/ResetException.java
#	src/test/java/fr/esgi/simple_auth_java/ManagerTest.java

TristanEduProjet committed:
update from rest/common

# Conflicts:
#	src/main/java/fr/esgi/simple_auth_java/Manager.java
#	src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordByMail.java
#	src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithOldPwd.java
#	src/main/java/fr/esgi/simple_auth_java/reset/ResetorPasswordWithTempPwdByMail.java

TristanEduProjet committed:
refresh from rest/common

# Conflicts:
#	src/main/java/fr/esgi/simple_auth_java/Manager.java

Commit:
Work in progress

TristanEduProjet committed:
Merge branch 'reset/common' into reset/test

Commit:
Work in progress

TristanEduProjet committed:
Merge branch 'reset/common' into reset/test

TristanEduProjet committed:
Correction of classes for unit tests

TristanEduProjet committed:
Implements tests of resetor
